### ISSUE TITLE
Release 0.8.0-alpha

### ISSUE THREAD

**Issue by johannes-b** [keptn/keptn#2807]
# Keptn 0.8.0-alpha

* **Link to milestone: [Milestone 0.8.0](https://github.com/keptn/keptn/milestone/17)**
* **Release Date: <= December 29 th, 2020**
* **Release Manager: Christian Kreuzberger (@christian-kreuzberger-dtx )**

### Step 1 - Release of keptn/spec

**Artifact:** GitHub Release `keptn/spec:0.2.0-alpha` (also referred as ``spec.keptn.sh/x.y.z``)

**Assignee**: @johannes-b 

**Repo**: https://github.com/keptn/spec

**Note**: Not every Keptn release requires a new Keptn spec.

- [x] All release-related PRs/discussions are resolved and are merged into master (please check back with the Issue/PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] *Draft* a new release https://github.com/keptn/spec/releases with release notes (collect them from PR/discussion titles) on master branch
- [x] Ask for feedback (e.g., on Slack)
- [x] **ONLY FOR RELEASE MANAGER**: Release notes approved
- [x] If you're happy with everything, pull the trigger and release the spec
- [x] **ONLY FOR RELEASE MANAGER**: keptn/spec:x.y.z released on GitHub
- [x] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)

---
### Step 2 - Release of keptn/examples

**Artifact:** GitHub Release `keptn/examples:0.8.0-alpha`

**Assignee**: @johannes-b 

**Repo**: https://github.com/keptn/examples

**Note**: Use the same version as for keptn/keptn (every Keptn release requires a new release of Keptn examples)

- [x] All release-related feature/bugfix/hotfix branches are merged into master (please check back with the PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Created `release-x.x.x` branch: `git checkout -b release-x.x.x`
- [x] _on release-x.x.x branch:_ releasenotes_develop.md renamed to releasenotes_**Vx.x.x**.md (**AUTOMATION WANTED**)
- [x] _on release-x.x.x branch:_ collect release-notes either based on GitHub Issues for this Release or via commit messages (**AUTOMATION WANTED**)
- [x] _on release-x.x.x branch:_ Update README.md with links to *Maintained Examples and Keptn Use cases* (not for beta releases)
- [x] _on release-x.x.x branch:_ `git commit`
- [x] Create PR from release-x.x.x to master and request reviews (review on release notes is mandatory)
- [x] **ONLY FOR RELEASE MANAGER**: Release notes approved
- [x] Once the PR is approved and merged into master, *draft* a new release https://github.com/keptn/examples/releases with the release notes on the `release-x.x.x` branch (**AUTOMATION WANTED**)
- [x] If you're happy with everything, pull the trigger and release the examples
- [x] **ONLY FOR RELEASE MANAGER**: keptn/examples:x.x.x released on GitHub
- [x] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)
- [x] Created new and empty release notes (releasenotes_develop.md) file on master branch

--- 
### Step 3 - Release of Keptn kubernetes-utils

**Artifact:** GitHub Release of `keptn/kubernetes-utils:v0.8.0-alpha` 

**Assignee**: @bacherfl 

**Repo**: https://github.com/keptn/kubernetes-utils

**Note:** The `v` in the version, which can be used in GoLang using `go get github.com/keptn/kubernetes-utils@vx.x.x`

- [x] All release-related feature/bugfix/hotfix branches are merged into master (please check back with the PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Created `release-x.x.x` branch: `git checkout -b release-x.x.x`
- [x] _on release-x.x.x branch:_ releasenotes_develop.md renamed to releasenotes_**Vx.x.x**.md
- [x] _on release-x.x.x branch:_ collect release-notes either based on GitHub Issues for this Release or via commit messages (**AUTOMATION WANTED**)
- [x] _on release-x.x.x branch:_ `git commit`
- [x] Create PR from release-x.x.x to master and request reviews (review on release notes is mandatory)
- [x] **ONLY FOR RELEASE MANAGER**: Release notes approved
- [x] Once the PR is approved and merged into master, *draft* a new release https://github.com/keptn/kubernetes-utils/releases with the release notes on the `release-x.x.x` branch - please note the prefix `v` in the version name (**AUTOMATION WANTED**)
- [x] If you're happy with everything, pull the trigger and release the examples
- [x] **ONLY FOR RELEASE MANAGER**: keptn/kubernetes-utils:vx.x.x released on GitHub
- [x] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)
- [x] Created new and empty release notes (releasenotes_develop.md) file on master branch

--- 
### Step 4 - Release of Keptn go-utils

**Artifact:** GitHub Release of `keptn/go-utils:v0.8.0-alpha` 

**Assignee**: @bacherfl 

**Repo**: https://github.com/keptn/go-utils

**Note:** The `v` in the version, which can be used in GoLang using `go get github.com/keptn/go-utils@vx.x.x`

- [x] All release-related feature/bugfix/hotfix branches are merged into master (please check back with the PR owner)
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Created `release-x.x.x` branch: `git checkout -b release-x.x.x`
- [x] _on release-x.x.x branch:_ version bump (**Is this needed?**)
- [x] _on release-x.x.x branch:_ releasenotes_develop.md renamed to releasenotes_**Vx.x.x**.md
- [x] _on release-x.x.x branch:_ collect release-notes either based on GitHub Issues for this Release or via commit messages (**AUTOMATION WANTED**)
- [x] _on release-x.x.x branch:_ `git commit`
- [x] Create PR from release-x.x.x to master and request reviews (review on release notes is mandatory)
- [x] **ONLY FOR RELEASE MANAGER**: Release notes approved
- [x] Once the PR is approved and merged into master, *draft* a new release https://github.com/keptn/go-utils/releases with the release notes on the `release-x.x.x` branch - please note the prefix `v` in the version name (**AUTOMATION WANTED**)
- [x] If you're happy with everything, pull the trigger and release the examples
- [x] **ONLY FOR RELEASE MANAGER**: keptn/go-utils:vx.x.x released on GitHub
- [x] Notify Developers on Slack that a new release has been created (**AUTOMATION WANTED**)
- [x] Created new and empty release notes (releasenotes_develop.md) file on master branch

## Release of Keptn

:warning: **WARNING/STOP/DANGERZONE**: Do not start with this section unless **all the above steps** have been completed successfully.

---
### Step 5 - Create Release Branch for keptn-contrib services

**Please update version tags of below services accordingly; use semantic versioning**

**foreach:**
- [x] dynatrace-service > **dynatrace-service:0.11.0-alpha** ***
- [x] dynatrace-sli-service > **dynatrace-sli-service:0.8.0-alpha** ***
- [x] prometheus-service > **prometheus-service:0.4.0-alpha** **
- [x] prometheus-sli-service > **prometheus-sli-service:0.3.0-alpha** **
- [ ] argo-service > **argo-service:0.2.0-alpha** *
- [ ] servicenow-service > **servicenow-service:x.x.x** **TBD** * 
 - [ ] notification-service > **notification-service:x.x.x** **TBD** *
- [x] unleash-service > **unleash-service:0.3.0-alpha** ***

follow https://github.com/keptn/keptn/wiki/Keptn-Contrib-Release-Checklist Step 1

---
### Step 6 - Release of Keptn

- [ ] `keptn upgrade` - verify if any changes/migrations/... in the CLI command are needed for this release
- [ ] ~~`keptn upgrade` - Create a PR at https://github.com/keptn/get.keptn.sh and modify [version.json](https://github.com/keptn/get.keptn.sh/blob/master/src/version.json) accordingly~~
- [ ] ~~Upgrade docs available in https://keptn.sh/docs/~~
- [x] Git submodule for *specification* directory in keptn release branch are updated (e.g., `cd specification && git fetch && git checkout a.b.c && cd .. && git add specification && git commit -m "updated spec to a.b.c" && git push`)
   - **Note**: In case the `specification` directory in the keptn/keptn repo is empty, execute the following command **before**: `git submodule update --init`.
- [x] All feature/bugfix/hotfix branches are merged into master
- [x] **Cleanup (Requires ADMIN role):** Inactive branches are deleted
- [x] Latest go-utils version is used (run: `go get github.com/keptn/go-utils@vx.x.x`) in all sub-folders that use it (see [this code snippet](https://github.com/keptn/go-utils/blob/247b98ef868c6234cf5cb5f852551c4746d472ea/.travis.yml#L56-L69) to use an example)
- [x] Latest kubernetes-utils version is used (run: `go get github.com/keptn/kubernetes-utils@vx.x.x`) in all sub-folders that use it: `api`, `cli`, `configuration-service`, `gatekeeper-service`, `helm-service`.
- [x] Keptn release notes are up-to-date and spec version is referenced in release notes
- [x] releasenotes_develop.md renamed to releasenotes_**Vx.x.x**.md
- [x] Created `release-x.x.x` branch: `git checkout -b release-x.x.x`
- [ ] Ask stakeholders to review the release notes (*Note:* Do not use a GitHub DRAFT Release, it does not allow editing from multiple parties at the same time and we will lose changes**)

Check on GCP Cloud Storage (Attention: This is not automated anymore): 
- [x] keptn-cli x.x.x  @christian-kreuzberger-dtx 
- [x] keptn-installer x.x.x  @christian-kreuzberger-dtx 

Notify certain users and stakeholders about the release, such that they can test it and ensure compatibility with their services/docs. Make sure to provide instructions on how to download the CLI and install Keptn. **Make sure to let them know that this might not the final release, rather that it is a release candidate.**

- [x] Notify developers, keptn/tutorials CODEOWNERS, keptn-contrib and keptn-sandbox repo owners via Keptn Slack (e.g., keptn-integrations channel) about the upcoming release and provide a preview to the release notes

---
### Step 7 - Preparation phase

- [ ] keptn-sockshop/carts released 
- [ ] ~~keptn.sh/docs/integrations/ is up-to-date~~
- [x] keptn CLI command docu generated and updated on keptn.sh/docs/x.x.x/reference/cli/commands/  @christian-kreuzberger-dtx 
- [ ] keptn.sh/docs/ is up-to-date
- [ ] ~~tutorials.keptn.sh: update of tutorials in this branch: xyz~~

---
### Step 8 - Test phase

Upgrader:
- [ ] Upgrade command tested for full installation (GKE or K3s)
- [ ] Upgrade command tested for quality-gates/control-plane installation (GKE or K3s)
- [ ] Upgrade command tested for full installation (OpenShift 3.11 or MiniShift)
- [ ] Upgrade command tested for quality-gates/control-plane installation (OpenShift 3.11 or MiniShift)

Tests:
- Download of **keptn CLI x.x.x** > [ToDo: LINK to CLI]()
- Install keptn cli and execute: `keptn version`
- For keptn-contrib services: follow https://github.com/keptn/keptn/wiki/Keptn-Contrib-Release-Checklist Step 2
- Pre-view of **keptn.sh/docs** > [ToDo: LINK]()
- [ ] GCP - GKE - **Keptn core team**
  - [ ] installation: `keptn install`
    - [ ] quality gates UC completed according to keptn.sh/docs with Dynatrace
    - [ ] quality gates UC completed according to keptn.sh/docs with Prometheus
  - [x] Installation: `keptn install --use-case=continuous-delivery`
    - [x] Verify that the correct images and tags are used: `kubectl get deployments -n keptn -o wide`
    - [ ] All tutorials completed with Dynatrace
        - [x] Onboarding a Service
        - [x] Deployments with Quality Gates
        - [ ] Up-scale with Dynatrace
        - [x] Self-healing with Feature Flags
    - [ ] All tutorials completed with Prometheus 
        - [x] Onboarding a Service
        - [x] Deployments with Quality Gates
        - [x] Up-scale with Prometheus
  - [ ] Bridge verified
  - [x] Delete a project used in the use cases: `keptn delete project`
  - [ ] Uninstall Keptn from cluster: `keptn uninstall`
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used K8s version
- [ ] OpenShift 3.11 - MiniShift
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used OpenShift version
- [ ] OpenShift 4.x
  - [ ] All tutorials completed with Dynatrace
      - [ ] Onboarding a Service
      - [ ] Deployments with Quality Gates
      - [ ] Self-healing with Feature Flags
  - [ ] Update https://keptn.sh/docs/x.x.x/operate/install/ based on used OpenShift version
  - [ ] Update support matrix on keptn.sh/docs and release notes
  
 ---
### Step 9 - ~~Official release of keptn-contrib services~~

**foreach:**
- [ ] dynatrace-service > **dynatrace-service:x.x.x**
- [ ] dynatrace-sli-service > **dynatrace-sli-service:x.x.x**
- [ ] prometheus-service > **prometheus-service:x.x.x**
- [ ] prometheus-sli-service > **prometheus-sli-service:x.x.x**
- [ ] argo-service > **argo-service:x.x.x**
- [ ] servicenow-service > **servicenow-service::x.x.x**
- [ ] notification-service > **notification-service:x.x.x**
- [ ] unleash-service > **unleash-service:x.x.x**

follow https://github.com/keptn/keptn/wiki/Keptn-Contrib-Release-Checklist Step 3

---
### Step 10 - Official release of Keptn

**Note**: Some steps are only for GA releases, not for alpha, beta, etc.
- [x] PR from release-x.x.x to master (**only if needed**)
- [ ] `keptn upgrade` - merge the PR at https://github.com/keptn/get.keptn.sh [version.json](https://github.com/keptn/get.keptn.sh/blob/master/src/version.json)

Check on GCP Cloud Storage: 
- [x] keptn-cli x.x.x
- [x] keptn-installer x.x.x

Release Keptn
- [x] Keptn on GitHub released based on release notes (**AUTOMATION WANTED**)
- [x] Make sure CLI binaries are added as assets to the release (**AUTOMATION WANTED**)
- [x] **release-x.x.x** branch tagged with **x.x.x** (**AUTOMATION WANTED**)
- [x] Merge PR from **release-x.x.x** to **master** (**merge only if needed**)
- [ ] Created new and empty release notes (releasenotes_develop.md) file on master branch

---
### Final steps
- [ ] Update **Quick Start** on keptn.sh with newest release (**only for GA, not for alpha/beta releases**)
- [ ] Update [get.keptn.sh repo](https://github.com/keptn/get.keptn.sh) (which auto-uploads to get.keptn.sh S3 bucket with) newest release (**only for GA, not for alpha/beta releases**)
- [ ] ~~Add release notes to: https://keptn.sh/docs/news/release_announcements/ (**AUTOMATION WANTED**)~~
- [ ] ~~Release of keptn.sh~~
- [ ] ~~Send out Slack/Twitter notification~~
- [ ] ~~Send out Release-Email~~
- [x] Make sure any (relevant) changes to the release checklist are also saved in the [wiki](https://github.com/keptn/keptn/wiki/Release-Checklist)
